ui/src/app/pages/admin/admin-user.tsx

```diff
@@ -1,179 +1,296 @@
-    this.setState({reloadingProfile: profile});
-    authDomainApi().updateUserDisabledStatus(
-      {email: profile.username, disabled: disable}).then(_ => {
-        this.reloadProfile(profile);
-      });
-  }
+interface State {
+  loading: boolean;
+  profile: Profile;
+  saveDisabled: boolean;
+  verifiedInstitutionOptions: Array<{label: string, value: {displayName: string, shortName: string}}>;
+}
 
-  // We want to sort first by beta access status, then by
-  // submission time (newest at the top), then alphanumerically.
-  sortProfileList(profileList: Array<Profile>): Array<Profile> {
-    return profileList.sort((a, b) => {
-      // put disabled accounts at the bottom
-      if (a.disabled && b.disabled) {
-        return this.timeCompare(a, b);
-      }
-      if (a.disabled) {
-        return 1;
-      }
-      if (!!a.betaAccessBypassTime === !!b.betaAccessBypassTime) {
-        return this.timeCompare(a, b);
-      }
-      if (!!b.betaAccessBypassTime) {
-        return -1;
-      }
-      return 1;
-    });
-  }
 
-  private timeCompare(a: Profile, b: Profile): number {
-    if (a.betaAccessRequestTime === b.betaAccessRequestTime) {
-      return this.nameCompare(a, b);
-    } else if (a.betaAccessRequestTime === null) {
-      return 1;
-    } else if (b.betaAccessRequestTime === null) {
-      return -1;
-    }
-    return b.betaAccessRequestTime - a.betaAccessRequestTime;
-  }
+const AdminUser = withUrlParams()(class extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
 
-  private nameCompare(a: Profile, b: Profile): number {
-    if (a.familyName === null) {
-      return 1;
-    }
-    if (a.familyName.localeCompare(b.familyName) === 0) {
-      if (a.givenName === null) {
-        return 1;
-      }
-      return a.givenName.localeCompare(b.givenName);
-    }
-    return a.familyName.localeCompare(b.familyName);
+    this.state = {
+      loading: false,
+      profile: null,
+      saveDisabled: true,
+      verifiedInstitutionOptions: []
+    };
   }
 
-  convertDate(date): string {
-    return new Date(date).toString().split(' ').slice(1, 5).join(' ');
-  }
+  async componentDidMount() {
+    const {gsuiteDomain} = serverConfigStore.getValue();
+    const profileResponse = await profileApi().getUserByUsername(this.props.urlParams.usernameWithoutGsuiteDomain + "@" + gsuiteDomain);
 
-  convertProfilesToFields(profiles: Profile[]) {
-    return profiles.map(p => ({...p, name: p.familyName + ', ' + p.givenName,
-      betaAccessRequestTime: this.convertDate(p.betaAccessRequestTime),
-      bypass: <AdminUserBypass profile={p}/>, disabled: p.disabled.toString(),
-      userLockout: <LockoutButton disabled={this.state.reloadingProfile === p}
-        profileDisabled={p.disabled}
-        onClick={() => this.updateUserDisabledStatus(!p.disabled, p)}/>}));
+    const institutionsResponse = await institutionApi().getInstitutions();
```

Comment:
Same nitpick here: it would be nice to have some sort of error message if the institutions API call fails for some reason.